internal/provider/data_source_gitlab_user_keys.go

```diff
@@ -0,0 +1,70 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_user_keys", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_user_keys`" + ` data source allows a list of SSH keys to be retrieved by either the user ID or username.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/users.html#list-ssh-keys-for-user)`,
+
+		ReadContext: dataSourceGitlabUserKeysRead,
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Description: "The user ID.",
+				Type:        schema.TypeInt,
+				Required:    true,
+			},
```

Comment:
🚧 **Suggestion:** The `user_id` doesn't match the description and the potential of the upstream API which accepts both an integer user id or a string username.

We could implement it similar to what we have in the `gitlab_users` data source already:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/6540e190e384a86c4fc61da32f86332db1b9d175/internal/provider/data_source_gitlab_user.go#L24

Reply:
Done